force-app/main/default/classes/GreetingNotification.cls

```diff
@@ -0,0 +1,66 @@
+// 4.    Create a scheduled process that runs every last day of the moth at 11 pm:
+// 4.1    Create task for “Best doctor” with congrats text “Congratulations! You are the best doctor of the month and will receive the bonus!” and populate Bonus field [“Payments for getting bonus” (from doctor field) * “Bonus rate” (from metadata settings)]
+// 4.2    Clear “Payments amount” and uncheck “Best doctor of the month” for all Doctor objects
+public with sharing class GreetingNotification implements Schedulable{
+    public void scheduleProcess(){
+        String cronExpression = '0 0 0 L * ?'; // Schedule to run on January 1st at midnight
+        System.schedule('GreetingNotification Job', cronExpression, this);
+    }
+
+    public void execute(SchedulableContext context){
+        notification();
+    }
+
+    public static void notification(){
+        List<Contact> bestDoctorsOfMonthList = [SELECT Id, Best_doctor_of_the_month__c, Email, Payments_amount__c, AccountId, Account.Department__c, Account.Department_type__c, Account.Doctor_payments_to_receive_bonus__c
+                                                FROM Contact
+                                                WHERE Best_doctor_of_the_month__c = TRUE];
+
+        List<String> doctorsEmailsList = new List<String>();
+
+        List<Department_Type_Department__mdt> departmentMetadata = Department_Type_Department__mdt.getAll().values();
+
+        List<Task> bonusPaymentsTask = new List<Task>();
+
+        for (Contact doctor_i : bestDoctorsOfMonthList){
+            List<Department_Type_Department__mdt> doctorsDepartmentMetadataList = [SELECT Department__r.Label, Department__r.Bonus_amount__c, Department_Type__r.Label, Department_Type__r.Bonus_rate__c
```

Comment:
But it hasn't values that i need. Because i have lookup for another custom metadata Department_mdt and Department_Type__mdt. And as i remember i don't have any limits on query custom metadata.